godotopenxrmeta/src/main/cpp/openxr_fb_scene_extension_wrapper.cpp

```diff
@@ -0,0 +1,162 @@
+/*                      https://godotengine.org                           */
+/**************************************************************************/
+/* Copyright (c) 2022-present Godot XR contributors (see CONTRIBUTORS.md) */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "include/openxr_fb_scene_extension_wrapper.h"
+
+#include <godot_cpp/variant/utility_functions.hpp>
+#include <godot_cpp/classes/open_xrapi_extension.hpp>
+#include <godot_cpp/classes/object.hpp>
+
+#include "include/openxr_fb_spatial_entity_extension_wrapper.h"
+
+using namespace godot;
+
+OpenXRFbSceneExtensionWrapper *OpenXRFbSceneExtensionWrapper::singleton = nullptr;
+
+OpenXRFbSceneExtensionWrapper *OpenXRFbSceneExtensionWrapper::get_singleton() {
+	if (singleton == nullptr) {
+		singleton = memnew(OpenXRFbSceneExtensionWrapper());
+	}
+	return singleton;
+}
+
+OpenXRFbSceneExtensionWrapper::OpenXRFbSceneExtensionWrapper() :
+		OpenXRExtensionWrapperExtension() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbSceneExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_SCENE_EXTENSION_NAME] = &fb_scene_ext;
+	singleton = this;
+}
+
+OpenXRFbSceneExtensionWrapper::~OpenXRFbSceneExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbSceneExtensionWrapper::_bind_methods() {
+	ClassDB::bind_static_method("OpenXRFbSceneExtensionWrapper", D_METHOD("get_singleton"), &OpenXRFbSceneExtensionWrapper::get_singleton);
```

Comment:
I wasn't aware of the `reigster_singleton(...)` approach; let's use that instead and update `OpenXRFbSceneCaptureExtensionWrapper` accordingly!

Reply:
@dsnopek What's the earlier that `Engine::get_singleton()->register_singleton(...)` can be used?

I tried invoking it during `MODULE_INITIALIZATION_LEVEL_CORE` and I kept getting crash errors because the `Engine` singleton was not available.

Reply:
Also, doing `Engine.get_singleton(...)` does not retrieve the singleton instance.

Reply:
Here's a draft PR for reference: https://github.com/GodotVR/godot_openxr_vendors/pull/72

Reply:
I'm getting segfaults when registering using this approach. Did a bit of digging, and I think it's due to Engine::get_singleton() not being available in a gdextension

```
ERROR: Failed to retrieve non-existent singleton 'Engine'.
   at: get_singleton_object (core/config/engine.cpp:270)
ERROR: Parameter "singleton_obj" is null.
   at: get_singleton (thirdparty/godot-cpp/gen/src/classes/engine.cpp:49)
```

Digging into the code, godot-cpp auto-generates bindings that eventually call into Engine::get_singleton_object(const StringName &p_name) and passes the class name (in this case "Engine"), but godot never actually registers Engine itself as a singleton in this form, within the godot/ source tree it's just a raw pointer, and since it doesn't extend Object it's not trivial to register like all the other Singleton instances. From godot/core/config/engine.cpp

```
Engine *Engine::singleton = nullptr;

Engine *Engine::get_singleton() {
	return singleton;
}
```

I think we'd need to address that in the engine before we can use Engine::get_singleton() from an extension? In contrast, main.cpp registers Performance as follows: ```engine->add_singleton(Engine::Singleton("Performance", performance));```

 Not sure if you encountered something similar in #73 @dsnopek

Reply:
@maunvz There was some logic missing in `register_types.cpp` that @dsnopek fixed in https://github.com/GodotVR/godot_openxr_vendors/pull/73/files#diff-aaccbe3504fdb5822961fde8149e985f5539985f9e60a4a31d804f3bf5feb5e4.

In the updated [`register_types.cpp`](https://github.com/GodotVR/godot_openxr_vendors/pull/73/files#diff-aaccbe3504fdb5822961fde8149e985f5539985f9e60a4a31d804f3bf5feb5e4), the singleton registration is initialized at the `MODULE_INITIALIZATION_LEVEL_SCENE` level and the minimum initialization level was also updated to `MODULE_INITIALIZATION_LEVEL_SCENE`. 
With those changes in place, registration via the engine works as expected.